wetmap/src/components/newModals/shopModal/index.tsx

```diff
@@ -21,12 +23,28 @@ export default function ShopModal(props: ShopModalProps) {
     if (selectedShop) {
       getItineraries(selectedShop.id);
     }
+    if (
+      (profile?.partnerAccount) &&
+      (selectedShop?.userId === profile.UserID)
+    ) {
+      setIsMyShop(true);
+    } else {
+      setIsMyShop(false);
+    }
   }, [selectedShop]);
 
   useEffect(() => {
     if (profile && profile.partnerAccount) {
       setIsPartnerAccount(true);
     }
+    if (
+      (profile?.partnerAccount) &&
+      (selectedShop?.userId === profile.UserID)
+    ) {
+      setIsMyShop(true);
+    } else {
+      setIsMyShop(false);
+    }
   }, []);
```

Comment:
actually the empty dependency array one was completely redundant since useEffect always runs on first render anyhow. So I moved the small bit that was only in the empty array useEffect into the other one and added profile to the array along with selectedShop.